src/input/dca.rs

```diff
@@ -48,6 +48,12 @@ async fn _dca(path: &OsStr) -> Result<Input, DcaError> {
 
     let reader = json_reader.into_inner().into_std().await;
 
+    #[cfg(feature = "serenity")]
+    let metadata: Metadata =
+        serenity::json::prelude::from_slice::<DcaMetadata>(raw_json.as_mut_slice())
+            .map_err(DcaError::InvalidMetadata)?
+            .into();
+    #[cfg(not(feature = "serenity"))]
```

Comment:
General feedback for all other driver/input uses: wouldn't it make more sense to use the simd-json library ourselves if that feature is selected? That way we can apply this to twilight too, and use `serenity::json::*` only where we interface with serenity in the gateway portions. Twilight v0.8 now exposes this via `twilight_gateway::shard::raw_message::Message` from what I can see.

Reply:
I somewhat planned for this, but I have 0 idea how twilight manages SIMD; I will not make a PR of something I'm completely unfamiliar with. If someone else wants to implement the SIMD feature for twilight, go for it.